web/src/realtime-info/reducer.js

```diff
@@ -0,0 +1,23 @@
+import { REMOVE_PEER_STATE, UPDATE_PEER_STATE } from './actions'
+// const defaultState = {
+//   isOpen: false,
+//   goalAddress: null,
+// }
+const defaultState = {}
+
+export default function(state = defaultState, action) {
+  const { payload, type } = action
+  switch (type) {
+    case UPDATE_PEER_STATE:
+      return {
+        ...state,
+        [payload.agent_pub_key]: payload
+      }
+    case REMOVE_PEER_STATE:
+      state = state
+      delete state[payload.agent_pub_key]
+      return state
```

Comment:
here's a bizarre one... 

immutability can be a tough one to achieve in javascript, here's a bit about what happens when you write
`state = state` 

in the case where `state` was a simple value, such as a `string`, a number, or a boolean, we can assign the value of that value to another variable and essentially `clone` it, from rust lingo. 

```js
let a = 'test'
let b = a
b = 'test2'
console.log(a) // -> 'test'
```

however, if the values are arrays, or objects, (complex data structures) it doesn't clone, it just sort of... borrows I guess. If you edit the newly assigned, you will also find that it changes the original. 
```js
let a = { test: 'test' }
let b = a
b.test = 'test2'
console.log(a) // -> { test: 'test2' }
```

when we do destructuring, however, we produce a new copy of the inner values, onto a brand new object in memory. 
```js
let a = { test: 'test' }
let b = { ...a }
b.test = 'test2'
console.log(a) // -> { test: 'test' }
```

it's ok to mutate our new copy, which `delete` does, it mutates it. 
so
```suggestion
let newState = { ...state }
delete newState[payload.agent_pub_key
return newState
```